resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/KubernetesSuite.scala

```diff
@@ -212,7 +212,9 @@ class KubernetesSuite extends SparkFunSuite
       driverPodChecker: Pod => Unit = doBasicDriverPodCheck,
       executorPodChecker: Pod => Unit = doBasicExecutorPodCheck,
       appArgs: Array[String] = Array.empty[String],
-      isJVM: Boolean = true ): Unit = {
+      isJVM: Boolean = true,
+      sparkConf: Option[SparkAppConf] = None,
+      appLoc: Option[String] = None): Unit = {
```

Comment:
`appLoc` is not very clear name (at least to me). I had to scroll down to see that it is used for `spark-app-locator`.
Maybe rename to `appLocatorOverride` ?!
If you like the idea then maybe the same for `sparkConf` -> `sparkConfOverride`